test/infrastructure/docker/internal/docker/loadbalancer.go

```diff
@@ -137,7 +140,7 @@ func (s *LoadBalancer) Create(ctx context.Context) error {
 }
 
 // UpdateConfiguration updates the external load balancer configuration with new control plane nodes.
-func (s *LoadBalancer) UpdateConfiguration(ctx context.Context) error {
+func (s *LoadBalancer) UpdateConfiguration(ctx context.Context, unsafeLoadBalancerConfigCM *corev1.ConfigMap) error {
```

Comment:
We could keep the whole configmap logic at the calling function and only pass a string as `unsafeLoadBalancerConfigTemplate` here. And use the template if it is `!= ""`

```suggestion
func (s *LoadBalancer) UpdateConfiguration(ctx context.Context, unsafeLoadBalancerConfigCM string) error {
```

This way all the logic is at the above function (getting configmap, configmap key).